src/nvim/tag.c

```diff
@@ -1019,8 +1017,9 @@ static void prepare_pats(pat_T *pats, int has_re)
         if (vim_strchr((char_u *)(p_magic ? ".[~*\\$" : "\\$"),
                 pats->head[pats->headlen]) != NULL)
           break;
-    if (p_tl != 0 && pats->headlen > p_tl)      /* adjust for 'taglength' */
-      pats->headlen = p_tl;
+    if (p_tl != 0 && pats->headlen > (size_t)p_tl) {  // adjust for "taglength"
```

Comment:
Given that the purpose of the Wconversion changes is to avoid conversion mistakes, I think also changing lengths to size_t is an unncessary risk, especially where we have to cast signed values like this which may then result in a bad comparison.

E.g. `:set tl=-1` is actually accepted. But that will cause this comparison to do something different than Vim, and then we have a time-consuming bug to track down.

Similar problem caused bugs like [this](https://github.com/neovim/neovim/commit/2a6c5bb0c4b03a9da81dae64d37c9912e448eaf0).

So, please leave the `size_t` changes out of the `-Wconversion` PRs except for cases that are more localized (not interacting with `p_foo` or other **user-facing** global values.